cdap-ui/app/hydrator/services/canvas-factory.js

```diff
@@ -106,6 +106,80 @@
       return finalConnections.map(function (conn) { delete conn.visited; return conn; });
     }
 
+    function orderConditionAlertErrorConnections(connections, conditionNodes, alertNodes, errorNodes) {
+      let swappedConnections = false;
+
+      // If graph has a condition node, then have to make sure/rearrange connections from that
+      // node so that the 'true' connection always comes before the 'false' one, otherweise
+      // it will look bad
+      if (conditionNodes.length > 0) {
+        angular.forEach(conditionNodes, (conditionNode) => {
+          let trueConnIndex = _.findIndex(connections, (conn) => conn.from === conditionNode && conn.condition === true);
+          let falseConnIndex = _.findIndex(connections, (conn) => conn.from === conditionNode && conn.condition === false);
+
+          if (trueConnIndex === -1 || falseConnIndex === -1 || trueConnIndex === falseConnIndex - 1) {
+            return;
+          }
+
+          let falseConn = connections.splice(falseConnIndex, 1)[0];
+          if (falseConnIndex < trueConnIndex) {
+            connections.splice(trueConnIndex, 0, falseConn);
+          } else {
+            connections.splice(trueConnIndex + 1, 0, falseConn);
+          }
+
+          swappedConnections = true;
+        });
+      }
+
+      let connectionsObj = orderAlertErrorConnections(connections, errorNodes, swappedConnections);
+      connectionsObj = orderAlertErrorConnections(connections, alertNodes, swappedConnections);
+
+
+      return connectionsObj;
+    }
+
+    function orderAlertErrorConnections(connections, alertOrErrorNodes, swappedConnections) {
+      let connObj = {
+        connections,
+        swappedConnections
+      };
+
+      if (!alertOrErrorNodes.length) {
+        return connObj;
+      }
+
+      angular.forEach(alertOrErrorNodes, (node) => {
+        let lastConnToThisNode = _.findLast(connections, conn => conn.to === node);
+        let lastConnToThisNodeIndex = connections.indexOf(lastConnToThisNode);
```

Comment:
I think Lodash has `findLastIndex` method. You should do `findLastIndex` first, and to get `lastConnToThisNode` simply do `connections[index]`. This way, you are only traversing the array once.